schemahcl/hcl.go

```diff
@@ -541,6 +538,49 @@ func findAttr(attrs []*Attr, k string) (*Attr, bool) {
 	return nil, false
 }
 
+func hclRefTokens(ref string) hclwrite.Tokens {
+	s := strings.Split(strings.ReplaceAll(ref, "$", ""), ".")
+	t := make([]*hclwrite.Token, 0, len(s))
+	for i, s := range s {
+		if i == 0 {
```

Comment:
Maybe add comments explaining the cases

Reply:
Root element in NewExpressionAbsTraversal